proxylib/proxylib_binary_memcached_test.go

```diff
@@ -0,0 +1,480 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	_ "gopkg.in/check.v1"
+
+	"github.com/cilium/cilium/proxylib/binarymemcached"
+	"github.com/cilium/cilium/proxylib/proxylib"
+	"github.com/cilium/cilium/proxylib/test"
+	_ "github.com/cilium/cilium/proxylib/testparsers"
+)
+
+var getHello = []byte{
```

Comment:
Why can't this file be in the memcached package? Import cycle? Can we move it to `memcached/binary/test`?

Reply:
That's because most test helpers are in main package, which is non-importable. We can factor out all test helpers to a library in the future, but I think it's out of scope of this PR.

Reply:
OK. This definitely needs to be improved but I agree that it is outside of the scope of this PR.